src/shared/port-provider.js

```diff
@@ -112,41 +112,14 @@ export class PortProvider {
         type: 'request',
       },
     ];
-  }
 
-  /**
-   * Check that data and origin matches the expected values.
-   *
-   * @param {object} options
-   *   @param {Message} options.allowedMessage - the `data` must match this
-   *     `Message`.
-   *   @param {string} options.allowedOrigin - the `origin` must match this
-   *     value. If `allowedOrigin` is '*', the origin is ignored.
-   *   @param {any} options.data - the data to be compared with `allowedMessage`.
-   *   @param {string} options.origin - the origin to be compared with
-   *     `allowedOrigin`.
-   */
-  _messageMatches({ allowedMessage, allowedOrigin, data, origin }) {
-    if (allowedOrigin !== '*' && origin !== allowedOrigin) {
-      return false;
-    }
-
-    return isMessageEqual(data, allowedMessage);
-  }
-
-  /**
-   * @param {'frameConnected'} eventName
-   * @param {(source: 'guest'|'sidebar', port: MessagePort) => void} handler - this handler
-   *   fires when a request for the host frame has been granted.
-   */
-  on(eventName, handler) {
-    this._emitter.on(eventName, handler);
+    this._listen();
   }
 
   /**
    * Initiate the listener of port requests by other frames.
```

Comment:
The documentation for `listen` needs to be moved to / combined with the existing constructor documentation.

Reply:
Agree.